### PR TITLE
Set provider config from ENV vars instead of plain text files

### PR DESCRIPTION
**Pull request by iamargus95**
- [x] Generate config from env vars.
- [x] Sets config correctly when multiple blocks of the same Provider are present.
- Set all types of config values
    - [x] String
    - [x] Bool
    - [x] []string
    - [x] map[string]string 
-------------------------------------------------------------------------
Here is the `config.yaml` I used while testing locally.
```yaml
slack:
  - id: recon
    slack_channel: $SLACK_CHANNEL_1
    slack_username: $SLACK_USER_1
    slack_webhook_url: $SLACK_WEBHOOK_1
    slack_threads: $SLACK_THREADS_1
  - id: recon1
    slack_channel: $SLACK_CHANNEL_2
    slack_username: $SLACK_USER_2
    slack_webhook_url: $SLACK_WEBHOOK_2
    slack_threads: $SLACK_THREADS_2
pushover:
  - id: $PUSHOVER_ID
    pushover_api_token: $PUSH_API_TOKEN
    pushover_devices: $PUSH_DEVICES
custom:
  - id: $CUSTOM_ID
    custom_headers: $CUSTOM_HEADERS
```

These are the export statements I used to set the ENV variables.
```bash
export SLACK_CHANNEL_1=slackChannel1
export SLACK_USER_1=slackUser1
export SLACK_WEBHOOK_1=slackWebhook1
export SLACK_THREADS_1=true
export SLACK_CHANNEL_2=slackChannel2
export SLACK_USER_2=slackUser2
export SLACK_WEBHOOK_2=slackWebhook2
export SLACK_THREADS_2=false
export PUSHOVER_ID=pushoverID
export PUSH_API_TOKEN=123qwerty123helloworld
export PUSH_DEVICES='["device1","device2","device3"]'
export CUSTOM_ID=customID
export CUSTOM_HEADERS='{"Content-Type":"application/json","X-Api-Key": "XXXXX"}'
```

### CODE REVIEW

**Comment by ShubhamRasal** [comment]
we can go with second approach..
```go
// utils.go
func readProviderConfig(filepath string) (io.Reader, error) {
	// Open the file
	file, err := os.Open(filepath)
	if err != nil {
		return nil, err
	}
	defer file.Close()

	// Create a scanner to read the file line by line
	scanner := bufio.NewScanner(file)

	// Create a string builder to accumulate the modified data
	var sb strings.Builder

	// Iterate over each line and do variable substitution
	for scanner.Scan() {
		line := scanner.Text()
		newLine := substituteEnvVars(line)
		sb.WriteString(newLine)
		sb.WriteString("\n")
	}
	// Check for errors
	if err := scanner.Err(); err != nil {
		return nil, err
	}

	return strings.NewReader(sb.String()), nil
}

func substituteEnvVars(line string) string {
	for _, word := range strings.Fields(line) {
		word = strings.Trim(word, " \"")
		if strings.HasPrefix(word, "$") {
			key := strings.TrimPrefix(word, "$")
			val := os.Getenv(key)
			line = strings.ReplaceAll(line, word, val)
		}
	}
	return line
}
```
`runner.go`
```go
	reader, err := readProviderConfig(options.ProviderConfig)
	if err != nil {
		return nil, errors.Wrap(err, "could not open provider config file")
	}
	if parseErr := yaml.NewDecoder(reader).Decode(&providerOptions); parseErr != nil {
		return nil, errors.Wrap(parseErr, "could not parse provider config file")
	}
```

**Review by tarunKoyalwar — CHANGES_REQUESTED** [review]
nice work @iamargus95 , lgtm ! verified locally and seems to work 
ex config
```console
cat ~/.config/notify/provider-config.yaml
discord:
  - id: "crawl"
    discord_channel: "crawl"
    discord_username: "test"
    discord_format: "{{data}}"
    discord_webhook_url: "$DISCORD_WEBHOOK_URL"

```

**Comment by tarunKoyalwar** [comment]
@iamargus95 , since this is a generic implementation what do you think of creating new PR to move this function to https://github.com/projectdiscovery/utils/tree/main/file . that way we can extend this support to other projects

Thanks

File: internal/runner/runner.go [hunk 1]
```diff
@@ -143,3 +143,46 @@ func (r *Runner) sendMessage(msg string) error {
...
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return strings.NewReader(sb.String()), nil
+}
+
+func substituteEnvVars(line string) string {
+	for _, word := range strings.Fields(line) {
+		word = strings.Trim(word, " \"")
```
[the remark this answers is not in the document]
Comment (iamargus95) [reply internal/runner/runner.go]: Done ✅ 😇

**Comment by iamargus95** [comment]
@tarunKoyalwar 
Kindly review this PR in utils.
I've added this functionality as a helper.
- https://github.com/projectdiscovery/utils/pull/149

**Review by tarunKoyalwar — CHANGES_REQUESTED** [review]
@iamargus95 , since upstream pr https://github.com/projectdiscovery/utils/pull/149 is merged now can you update this PR to use those helper utils

**Comment by iamargus95** [comment]
> @iamargus95 , since upstream pr [projectdiscovery/utils#149](https://github.com/projectdiscovery/utils/pull/149) is merged now can you update this PR to use those helper utils

@tarunKoyalwar I think the latest release (v0.0.30) does not have the newly added helper function. Waiting for it to be added.

**Comment by ehsandeep** [comment]
> > @iamargus95 , since upstream pr [projectdiscovery/utils#149](https://github.com/projectdiscovery/utils/pull/149) is merged now can you update this PR to use those helper utils
> 
> @tarunKoyalwar I think the latest release (v0.0.30) does not have the newly added helper function. Waiting for it to be added.

https://github.com/projectdiscovery/utils/releases/tag/v0.0.31

**Pull request merged**